src/MySqlConnector/MySqlClient/Caches/CachedProcedure.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Data;
+using System.Threading;
+using System.Threading.Tasks;
+using MySql.Data.Protocol.Serialization;
+
+namespace MySql.Data.MySqlClient.Caches
+{
+	internal class CachedProcedure
+	{
+		internal static async Task<CachedProcedure> FillAsync(IOBehavior ioBehavior, MySqlConnection connection, string schema, string component, CancellationToken cancellationToken)
+		{
+			var cmd = (MySqlCommand) connection.CreateCommand();
+
+			cmd.CommandText = @"SELECT ORDINAL_POSITION, PARAMETER_MODE, PARAMETER_NAME, DATA_TYPE, DTD_IDENTIFIER
+				FROM information_schema.parameters
+				WHERE SPECIFIC_SCHEMA = @schema AND SPECIFIC_NAME = @component
+				ORDER BY ORDINAL_POSITION";
+			cmd.Parameters.Add(new MySqlParameter
+			{
+				ParameterName = "@schema",
+				DbType = DbType.String,
+				Value = schema
+			});
+			cmd.Parameters.Add(new MySqlParameter
+			{
+				ParameterName = "@component",
+				DbType = DbType.String,
+				Value = component
+			});
+
+			var parameters = new List<CachedParameter>();
+			using (var reader = (MySqlDataReader) await cmd.ExecuteReaderAsync(CommandBehavior.Default, ioBehavior, cancellationToken).ConfigureAwait(false))
+			{
+				while (await reader.ReadAsync(cancellationToken).ConfigureAwait(false))
+				{
+					parameters.Add(new CachedParameter(
+						reader.GetInt32(0),
+						!reader.IsDBNull(1) ? reader.GetString(1) : null,
+						!reader.IsDBNull(2) ? reader.GetString(2) : null,
+						reader.GetString(3),
+						reader.GetString(4).ToLower().Contains("unsigned")
```

Comment:
`reader.GetString(4).IndexOf("unsigned", StringComparison.OrdinalIgnoreCase) != -1`

Avoids culture-sensitive comparison ("Turkish I") and allocating a new string just for substring comparison.